docker/compose/docker-compose.mariadb.yml

```diff
@@ -39,11 +39,12 @@ services:
     restart: unless-stopped
     volumes:
       - dbdata:/var/lib/mysql
+    env_file: docker-compose.env
     environment:
       MARIADB_HOST: paperless
-      MARIADB_DATABASE: paperless
-      MARIADB_USER: paperless
-      MARIADB_PASSWORD: paperless
+      MARIADB_DATABASE: $${PAPERLESS_DBNAME}
```

Comment:
yes you're right. i had no problems for existing container, but it don't work for new containers.

you need single dollar but the docker-compose.env needs to be renamed from docker-compose.env to .env
see -> https://stackoverflow.com/questions/73449390/docker-compose-custom-env-file-unexpected-behaviour

Reply:
Yes, single dollar it is.

I have everything in a single environment file, literally named `.env` (contrary to the two separate environment files paperless-ngx suggests in its default setup) as it's the only one Docker Compose loads by default.

I do this with every software I'm self-hosting and never had to deal with "Can't connect to database" errors again because the database password is always correct :)

<details><summary><code>.env</code></summary>

```bash
PAPERLESS_DBHOST=db
PAPERLESS_DBNAME=paperless
PAPERLESS_DBUSER=paperless
PAPERLESS_DBPASS=paperless
```
</details>

<details><summary><code>docker-compose.yml</code></summary>

```yaml
services:
  app:
    image: ghcr.io/paperless-ngx/paperless-ngx:latest
    # ...
    env_file: .env

  db:
    image: postgres:15
    # ...
    environment:
      POSTGRES_DB: ${PAPERLESS_DBNAME}
      POSTGRES_USER: ${PAPERLESS_DBUSER}
      POSTGRES_PASSWORD: ${PAPERLESS_DBPASS}
```
</details>